src/foam/nanos/auth/AgentAuthService.js

```diff
@@ -42,6 +42,32 @@ foam.INTERFACE({
         }
       ]
     },
+    {
+      name: 'canActAs',
+      documentation: `Returns a boolean indicating whether the provided agent has adequate permission to actAs the provided entity.`,
+      javaReturns: 'boolean',
+      swiftReturns: 'Bool',
+      javaThrows: [
+        'foam.nanos.auth.AuthorizationException',
+        'foam.nanos.auth.AuthenticationException'
+      ],
+      swiftThrows: true,
+      args: [
+        {
+          name: 'x',
+          javaType: 'foam.core.X',
+          swiftType: 'Context'
+        },
+        {
+          name: 'agent',
```

Comment:
I would like better names for agent and entity.

Reply:
So would I, but I think that's a different issue than what's being addressed here.

Reply:
I don't think so. Name discussion is a suitable topic for code reviews.

Reply:
Discussed offline. Name changes will be addressed in a different PR.